ddtrace/internal/sma.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import division
+
+DEFAULT_SMA_WINDOW = 10
+
+
+class SimpleMovingAverage(object):
+    """
+    Simple Moving Average implementation.
+    """
+
+    __slots__ = (
+        "size",
+        "index",
+        "counts",
+        "totals",
+        "sum_count",
+        "sum_total",
+    )
+
+    def __init__(self, size=None):
+        """
+        Constructor for SimpleMovingAverage.
+
+        :param size: The size of the window to calculate the moving average.
+        :type size: :obj:`int`
+        """
+        self.index = 0
+        self.size = size is None and DEFAULT_SMA_WINDOW or max(1, size)
```

Comment:
I'd rather raise a ValueError if size < 1 than silently changing it to 1, better to be loud about these kind of errors 🙂 

Also this is a little hard to read. It is correct but it's not immediately obvious what this is doing which is really just

```python
self.size = DEFAULT_SMA_WINDOW if size is None else size
```